superset/security/manager.py

```diff
@@ -1858,9 +1858,13 @@ def raise_for_access(
                 or self.can_access("datasource_access", datasource.perm or "")
                 or self.is_owner(datasource)
                 or (
+                    # Check whether the datasource is associated with a dashboard in a
+                    # trustworthy manner, i.e., we need to validate that the specified
+                    # dashboard is actually associated with said datasource.
```

Comment:
I don't think I understand this logic completely, but if `dashboardId` can be spoofed in the form what's preventing users from reading any datasource?

Reply:
There's validation in the lines below that:
a. The user has access the the dashboard, and
b. The datasource being requested is present on the dashboard